test/server/admin/server_info_handler_test.cc

```diff
@@ -60,6 +60,42 @@ TEST_P(AdminInstanceTest, Memory) {
                                   Property(&envoy::admin::v3::Memory::total_thread_cache, Ge(0))));
 }
 
+// TODO(lambdai): Enable the test after surfacing the memory_shrink_succeed stats.
+#ifdef TCMALLOC
+TEST_P(AdminInstanceTest, DISABLED_MemoryShrinkAttemptStats) {
+  {
+    // Trigger stats creation.
+    Http::TestResponseHeaderMapImpl response_headers;
+    Buffer::OwnedImpl response;
+    EXPECT_EQ(Http::Code::OK, getCallback("/memory", response_headers, response));
+  }
+  {
+    Http::TestResponseHeaderMapImpl response_headers;
+    std::string body;
+    EXPECT_EQ(Http::Code::OK, admin_.request("/stats", "GET", response_headers, body));
+    EXPECT_THAT(body, HasSubstr("memory_shrink_attempt:"));
```

Comment:
@jmarantz  I can only find `http.admin.*` stats in this test but when I run `envoy envoy.memory_shrink_succeed` shows up as in this test. How can I modify this test to make this test case passing?
```
Expected: has substring "memory_shrink_attempt:"
  Actual: "{\"stats\":[{\"value\":0,\"name\":\"http.admin.downstream_cx_active\"},{\"value\":0,\"name\":\"http.admin.downstream_cx_delayed_close_timeout\"},{\"value\":0,\"name\":\"http.admin.downstream_cx_destroy\"},{\"value\":0,\"name\":\"http.admin.downstream_cx_destroy_active_rq\"},{\"value\":0,\"name\":\"http.admin.downstream_cx_destroy_local\"},{\"value\":0,\"name\":\"http.admin.downstream_cx_destroy_local_active_rq\"},{\"value\":0,\"name\":\"http.admin.downstream_cx_destroy_remote\"},{\"value\":0,\"name\":\"http.admin.downstream_cx_destroy_remote_active_rq\"},{\"value\":0,\"name\":\"http.admin.downstream_cx_drain_close\"},{\"value\":0,\"name\":\"http.admin.downstream_cx_http1_active\"},{\"value\":0,\"name\":\"http.admin.downstream_cx_http1_total\"},{\"value\":0,\"name\":\"http.admin.downstream_cx_http2_active\"},{\"value\":0,\"name\":\"http.admin.downstream_cx_http2_total\"},{\"value\":0,\"name\":\"http.admin.downstream_cx_http3_active\"},{\"value\":0,\"name\":\"http.admin.downstream_cx_http3_total\"},{\"value\":0,\"name\":\"http.admin.downstream_cx_idle_timeout\"},{\"value\":0,\"name\":\"http.admin.downstream_cx_max_duration_reached\"},{\"value\":0,\"name\":\"http.admin.downstream_cx_overload_disable_keepalive\"},{\"value\":0,\"name\":\"http.admin.downstream_cx_protocol_error\"},{\"value\":0,\"name\":\"http.admin.downstream_cx_rx_bytes_buffered\"},{\"value\":0,\"name\":\"http.admin.downstream_cx_rx_bytes_total\"},{\"value\":0,\"name\":\"http.admin.downstream_cx_ssl_active\"},{\"value\":0,\"name\":\"http.admin.downstream_cx_ssl_total\"},{\"value\":0,\"name\":\"http.admin.downstream_cx_total\"},{\"value\":0,\"name\":\"http.admin.downstream_cx_tx_bytes_buffered\"},{\"value\":0,\"name\":\"http.admin.downstream_cx_tx_bytes_total\"},{\"value\":0,\"name\":\"http.admin.downstream_cx_upgrades_active\"},{\"value\":0,\"name\":\"http.admin.downstream_cx_upgrades_total\"},{\"value\":0,\"name\":\"http.admin.downstream_flow_control_paused_reading_total\"},{\"value\":0,\"name\":\"http.admin.downstream_flow_control_resumed_reading_total\"},{\"value\":0,\"name\":\"http.admin.downstream_rq_1xx\"},{\"value\":0,\"name\":\"http.admin.downstream_rq_2xx\"},{\"value\":0,\"name\":\"http.admin.downstream_rq_3xx\"},{\"value\":0,\"name\":\"http.admin.downstream_rq_4xx\"},{\"value\":0,\"name\":\"http.admin.downstream_rq_5xx\"},{\"value\":0,\"name\":\"http.admin.downstream_rq_active\"},{\"value\":0,\"name\":\"http.admin.downstream_rq_completed\"},{\"value\":0,\"name\":\"http.admin.downstream_rq_http1_total\"},{\"value\":0,\"name\":\"http.admin.downstream_rq_http2_total\"},{\"value\":0,\"name\":\"http.admin.downstream_rq_http3_total\"},{\"value\":0,\"name\":\"http.admin.downstream_rq_idle_timeout\"},{\"value\":0,\"name\":\"http.admin.downstream_rq_max_duration_reached\"},{\"value\":0,\"name\":\"http.admin.downstream_rq_non_relative_path\"},{\"value\":0,\"name\":\"http.admin.downstream_rq_overload_close\"},{\"value\":0,\"name\":\"http.admin.downstream_rq_response_before_rq_complete\"},{\"value\":0,\"name\":\"http.admin.downstream_rq_rx_reset\"},{\"value\":0,\"name\":\"http.admin.downstream_rq_timeout\"},{\"value\":0,\"name\":\"http.admin.downstream_rq_too_large\"},{\"value\":0,\"name\":\"http.admin.downstream_rq_total\"},{\"value\":0,\"name\":\"http.admin.downstream_rq_tx_reset\"},{\"value\":0,\"name\":\"http.admin.downstream_rq_ws_on_non_ws_route\"},{\"value\":0,\"name\":\"http.admin.rs_too_large\"}]}"
```

Reply:
I looks at the text you pasted and didn't see 'memory_shrink_attempt' anywhere. Are you sure in this test you are hitting the code in question?

Reply:
that being said, for some reason it looks like your /stats request is in json mode, though I don't why that would be, and you were searching for "memory_shrink_attempt:" which is the text-mode format, not the json format.